fast64_internal/f3d/f3d_gbi.py

```diff
@@ -3238,18 +3238,23 @@ def spc(x):
 
 
 # A palette is just a RGBA16 texture with width = 1.
+@dataclass
 class FImage:
-    def __init__(self, name, fmt, bitSize, width, height, filename):
-        self.name = name
-        self.fmt = fmt
-        self.bitSize = bitSize
-        self.width = width
-        self.height = height
+    name: str
+    fmt: str
+    bitSize: str
+    width: int
+    height: int
+    filename: str
+    data: bytearray = field(init=False, compare=False)
+    startAddress: int = field(init=False, compare=False)
+    isLargeTexture: bool = field(init=False, compare=False)
+    converted: bool = field(init=False, compare=False)
+
+    def __post_init__(self):
+        self.converted = self.isLargeTexture = False
         self.startAddress = 0
         self.data = bytearray(0)
```

Comment:
```suggestion
    data: bytearray = field(init=False, compare=False, default_factory=bytearray)
    startAddress: int = field(init=False, compare=False, default=0)
    isLargeTexture: bool = field(init=False, compare=False, default=False)
    converted: bool = field(init=False, compare=False, default=False)
```

Reply:
Honestly these should have init true, I will modify the suggestion and commit it

Reply:
Btw your code would not use the default, the reason I had it like this was so the values would get initialized but they couldn´t be passed to the constructor

Reply:
feel free to change

fwiw afaict my code would have worked?
for example:
```py
from dataclasses import dataclass, field
@dataclass
class Thing:
    foo = field(init=False, default="bar")
print(Thing())  # Thing(foo='bar')
Thing("baz")  # TypeError: Thing.__init__() takes 1 positional argument but 2 were given
```
, it seems to work as expected

Reply:
> If init is true, the field will be a parameter to the class's __init__() function.
I guess this doesn´t imply anything about if the value will be initialized, thanks